File: test-websocket.py
#!/usr/bin/env python3

import asyncio
import websockets
import json
import sys

async def test_websocket():
    try:
        print("Testing WebSocket connection to ws://localhost:8900...")
        
        async with websockets.connect("ws://localhost:8900") as websocket:
            print("✅ WebSocket connection opened successfully!")
            
            # Test a simple subscription request
            subscribe_request = {
                "jsonrpc": "2.0",
                "id": 1,
                "method": "signatureSubscribe",
                "params": [
                    "5VERv8NMvzbJMEkV8xnrLkEaWRtSz9CosKDYlCJjBRCN8FHXvVSs8h7oprNJfj6gJV26pEgJZNMAUh2tCgKHU9Sy"
                ]
            }
            
            print("Sending subscription request...")
            await websocket.send(json.dumps(subscribe_request))
            
            # Wait for response with timeout
            try:
                response = await asyncio.wait_for(websocket.recv(), timeout=3.0)
                print(f"📨 Received: {response}")
            except asyncio.TimeoutError:
                print("⏰ No response received within 3 seconds")
            
    except Exception as e:
        print(f"❌ WebSocket error: {e}")
        return False
    
    return True

if __name__ == "__main__":
    result = asyncio.run(test_websocket())
    sys.exit(0 if result else 1)

File: test-websocket.js
#!/usr/bin/env node

const WebSocket = require('ws');

console.log('🔍 Testing Solana WebSocket connectivity...');

const ws = new WebSocket('ws://localhost:8900');

ws.on('open', function open() {
    console.log('✅ WebSocket connected successfully!');
    
    // Test a simple getHealth subscription
    const subscribeRequest = {
        jsonrpc: "2.0",
        id: 1,
        method: "getHealth"
    };
    
    console.log('📤 Sending test request:', JSON.stringify(subscribeRequest));
    ws.send(JSON.stringify(subscribeRequest));
    
    setTimeout(() => {
        console.log('✅ WebSocket test completed successfully');
        ws.close();
    }, 2000);
});

ws.on('message', function message(data) {
    console.log('📨 Received:', data.toString());
});

ws.on('error', function error(err) {
    console.error('❌ WebSocket error:', err.message);
});

ws.on('close', function close(code, reason) {
    console.log('🔌 WebSocket connection closed:', code, reason.toString());
    process.exit(0);
});

// Timeout after 5 seconds
setTimeout(() => {
    console.error('⏰ WebSocket connection timeout');
    ws.close();
    process.exit(1);
}, 5000);